File: Countries/shortest_country/shortest_country.py
# toy program to find the country with the shortest name

# importing the country list
countries = open("/home/evergreen/UnderDogsAssignments/Countries/countries.txt")

#turning countries.doc into an array
countries_list = []
for line in countries:
    countries_list.append(line.strip())

# Find the shortest country name  in the list of countries_list
shortest = 'United States of America'
length = len(shortest)
for country in countries_list:
    if len(country) < length:
        shortest = country
        length = len(shortest)

print(str(shortest) + " is the shortest country name")

File: Countries/shortest_country/shortestCountry.js
// Program to find the shortest string
// Importing the fs module
let fs = require("fs");

// Intitializing the readFileLines with the file
const readFileLines = filename =>
   fs.readFileSync(filename)
   .toString('UTF8')
   .split('\n');

// Calling the readFiles function with file name
let countries = readFileLines('/home/evergreen/UnderDogsAssignments/Countries/countries.txt');

let shortestCountry = countries[0];
for (let j = 0; j < countries.length - 1; j++) {
    if (countries[j].length < shortestCountry.length) {
        shortestCountry = countries[j];
    }
}

console.log(shortestCountry);
